app/models/user.rb

```diff
@@ -3,7 +3,7 @@ class User < ApplicationRecord
   has_many :comments, dependent: :destroy, foreign_key: :author_id
   has_many :likes, dependent: :destroy, foreign_key: :author_id
   validates :name, presence: true
-  #validates :posts_counter, numericality: { greater_than_or_equal_to: 0 }
+  # validates :posts_counter, numericality: { greater_than_or_equal_to: 0 }
 
   def top_three_recent_posts
     posts.includes(:author).order(created_at: :DESC).limit(3)
```

Comment:
https://github.com/ismailakinkunmi/blog/blob/405f87f87909824be07abd5fa725b68118b269a2/app/models/post.rb#L11-L13

- [ OPTIONAL The same thing inside for the on line 12, Calling post.comments.count causes a SQL COUNT statement to be executed, which defeats the purpose of the refactoring we’ve done so far. Using `author.posts.size` instead, will leverage the fact that we have already loaded our comments from the database (thanks to includes) and will simply count the records in memory. This way, we can keep the number of SQL statements needed to list all posts and their comments count down to 2.

- For more documentation, you can visit this [.sise Vs .count](https://medium.com/idopterlabs/solving-the-n-1-problem-in-rails-b5ea677e78e0#:~:text=Notice%20that%20in%20our%20new%20View%20code%20we%20are%20now%20using%20size%20instead%20of%20count.%20You%20might%20be%20thinking%20these%20two%20methods%20are%20interchangeable%20because%20they%20return%20the%20same%20thing.%20Not%20exactly!)